### ISSUE TITLE
Template parsing error (3.1.25)

### ISSUE THREAD

**Issue by abrookbanks** [smarty-php/smarty#60]
The following block of code:

```
                  {foreach from=$CUSTOM_QUICK_TASKS item=$task}
                  <li><a href="{$task.url}">{$task.name}</a></li>
                  {foreachelse}
                  <li><a href="?_g=reports&report[date][from]={$QUICK_TASKS.today}&report[date][to]={$QUICK_TASKS.today}">{$LANG.dashboard.task_orders_view_day}</a></li>
                  <li><a href="?_g=reports&report[date][from]={$QUICK_TASKS.this_weeks}&report[date][to]={$QUICK_TASKS.today}">{$LANG.dashboard.task_orders_view_week}</a></li>
                  <li><a href="?_g=reports">{$LANG.dashboard.task_orders_view_month}</a></li>
                  <li><a href="?_g=products&action=add">{$LANG.catalogue.product_add}</a></li>
                  <li><a href="?_g=categories&action=add">{$LANG.catalogue.category_add}</a></li>
                  {/foreach}
```
Is causing a PHP error:

> PHP Parse error:  syntax error, unexpected '$_smarty_tpl' (T_VARIABLE)

The PHP code generated is:

Notice ```$foreach_$_smarty_tpl->tpl_vars['task']```

```
<?php
$_from = $_smarty_tpl->tpl_vars['CUSTOM_QUICK_TASKS']->value;
if (!is_array($_from) && !is_object($_from)) {
settype($_from, 'array');
}
$_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value] = new Smarty_Variable;
$_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value]->_loop = false;
foreach ($_from as $_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value]->value) {
$_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value]->_loop = true;
$foreach_$_smarty_tpl->tpl_vars['task']->value_Sav = $_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value];
?>
                  <li><a href="<?php echo $_smarty_tpl->tpl_vars['task']->value['url'];?>
"><?php echo $_smarty_tpl->tpl_vars['task']->value['name'];?>
</a></li>
                  <?php
$_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value] = $foreach_$_smarty_tpl->tpl_vars['task']->value_Sav;
}
if (!$_smarty_tpl->tpl_vars[$_smarty_tpl->tpl_vars['task']->value]->_loop) {
?>
                  <li><a href="?_g=reports&report[date][from]=<?php echo $_smarty_tpl->tpl_vars['QUICK_TASKS']->value['today'];?>
&report[date][to]=<?php echo $_smarty_tpl->tpl_vars['QUICK_TASKS']->value['today'];?>
"><?php echo $_smarty_tpl->tpl_vars['LANG']->value['dashboard']['task_orders_view_day'];?>
</a></li>
                  <li><a href="?_g=reports&report[date][from]=<?php echo $_smarty_tpl->tpl_vars['QUICK_TASKS']->value['this_weeks'];?>
&report[date][to]=<?php echo $_smarty_tpl->tpl_vars['QUICK_TASKS']->value['today'];?>
"><?php echo $_smarty_tpl->tpl_vars['LANG']->value['dashboard']['task_orders_view_week'];?>
</a></li>
                  <li><a href="?_g=reports"><?php echo $_smarty_tpl->tpl_vars['LANG']->value['dashboard']['task_orders_view_month'];?>
</a></li>
                  <li><a href="?_g=products&action=add"><?php echo $_smarty_tpl->tpl_vars['LANG']->value['catalogue']['product_add'];?>
</a></li>
                  <li><a href="?_g=categories&action=add"><?php echo $_smarty_tpl->tpl_vars['LANG']->value['catalogue']['category_add'];?>
</a></li>
                  <?php
}
?>
```

**Comment by uwetews** [comment]
If you use the Smarty 2 style of {foreach} the attributes of item and key must be a string with the variable name, not the variable itself,

{foreach from=$from key='bar' item='buh'}

In your case {foreach from=$CUSTOM_QUICK_TASKS item='task'}

See http://www.smarty.net/docs/en/language.function.foreach.tpl

**Comment by abrookbanks** [comment]
Thanks we will try this.

**Comment by uwetews** [comment]
The original problem was cause by an invalid Smarty syntax 

´´´php
{foreach from=$CUSTOM_QUICK_TASKS item=$task}
´´´

But I decided to modify the {foreach} compiler to accept a variable at the key and item attribute in addition to a variable name string,

The patch is now in dev-master and will later be included in 3.1.28

**Comment by AnrDaemon** [comment]
I'd rather see Smarty2 syntax go away. This is year 2015 and we're few months away from PHP7.
